compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -5384,6 +5383,10 @@ private BLangLiteral createSimpleLiteral(Node literal, SyntaxKind sign, boolean
                 }
             }
 
+            if (type == SyntaxKind.IDENTIFIER_TOKEN && text.startsWith(IDENTIFIER_LITERAL_PREFIX)) {
```

Comment:
Isnt this a bug when transforming `transformEnumMember`. Shouldnt we do something like "transform(member.identifier())"

Reply:
No, This is a kind of a general issue since if we r to create a string literal from identifier we should get rid of the initial quote if present. `transformEnumMember` is the only place that calls `createSimpleLiteral` with an identifier at the moment.

Reply:
I also checked this previously.  Basically, we have to handle this in two places. 
1. createIdentifier() 2. createSimpleLiteral()
We do the unescaping also separately in the two cases.